modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -405,3 +421,60 @@ ModularGapConductanceConstraint::computeQpResidual(Moose::MortarType
   mooseError("We should never get here. We should have errored in the constructor.");
 #endif
 }
+
+void
+ModularGapConductanceConstraint::deduceGeometryParameters()
+{
+  Point position;
+  Real area = 0.0;
+  const auto my_pid = _communicator.rank();
+
+  // build side element list as (element, side, id) tuples
+  const auto bnd = _mesh.buildActiveSideList();
+
+  std::unique_ptr<const Elem> side_ptr;
+  for (const auto & tuple : bnd)
+    if (std::get<2>(tuple) == _primary_id)
+    {
+      const auto * elem = _mesh.elemPtr(std::get<0>(tuple));
+      if (elem->processor_id() != my_pid)
+        continue;
+
+      const auto side = std::get<1>(tuple);
+
+      // update side_ptr
+      elem->side_ptr(side_ptr, side);
+
+      // area of the (linearized) side
+      const auto side_area = side_ptr->volume();
```

Comment:
Does taking into account the `_coord`, e.g. `2*pi*r` or `4*pi*r^2` matter for this? I'm guessing you actively don't want to take that into account, but wanted to check

Reply:
This routine should only be run for cartesian coordinate systems